style/ruby/README.md

```diff
@@ -3,12 +3,11 @@ Ruby
 
 [Sample](sample.rb)
 
-* Avoid conditional modifiers (lines that end with conditionals).
```

Comment:
I would suggest to go with @switzersc's suggestion (except with non-nested guard clauses or something like that). For example, IMO,

```ruby
# bad 

def do_something
  do_it if some_condition
end

# better

def do_something
  if some_condition
    do_it
  end
end
```

With that said, I do see some utility in guard clauses in more complex situations:

```ruby

def do_something
  return if user_is_tired?

  if hungry
    eat
  else
    have_fun
  end
end
```

I think one thing @georgebrock was getting at in his suggested code samples is that the usage of guard clauses often indicates some code design smells, which seems reasonable, and is valuable food for thought. With that said, I think we should embrace the reality that code is often not well designed, and guard clauses do provide a way to "semantically" distinguish between different types of conditionals (even though those semantics sometimes indicate a larger design pattern).